Update pods to support latest Authenticator release (1.16.0-beta.5)

mindgraffiti opened the pull request:
### Summary
In this PR, the latest Authenticator version is supported. The only pods updated are interdependencies that were required to update before Auth could be used. 

The latest Authenticator version removes the segues from Interface Builder and introduces a few convenience methods for loading storyboards and viewControllers. This will allow host apps to navigate freely between the viewControllers, using a navController instead of the segue / `prepare for segue` combination (should the host app's login flows change).

#### Major updates to a pod
Gridicons had a major version update, to version 1.0. There are method deprecations but no breaking changes. 

#### Minor updates to pods
- WordPressUI `1.5.0` -> `1.6.0`
- WordPressKit `4.5.4` -> `4.8.0`
- Alamofire `4.7.3` -> `4.8.0`
- WordPressAuthenticator `1.10.4` -> `1.16.0-beta.5`
- AlamofireImage `3.4.1` -> `3.5.2`

#### Patch version updates to pods
Many of the pods that updated their patch versions were conversions from Swift 4.2 to Swift 5 and/or allowed compatibility to build with Xcode `11.2.1`:
- NSObject-SafeExpectations from `0.0.3` -> `0.0.4`
- NSURL+IDN `0.3` -> `0.4`
- Lottie `2.5.2` -> `3.1.6`
- wpxmlrpc `0.8.4` -> `0.8.5`

### To test
1. Delete the app from the simulator
2. Clean the project
3. `bundle exec pod install` (no errors should appear in the terminal)
4. Build and run
5. Smoke test the app
**Expected:** no new warnings and no crashes.

#### Known bugs
A crash was found in develop but is also reproducible in this branch. (Let me know if I should file a bug or if this is a known issue.)

Steps to reproduce:
1. Posts > choose a post.
2. Do not make edits to the post. Select Cancel > Discard Changes.
The app crashes.

```
2020-05-18 15:44:39.603 [info][tid:main][RCTCxxBridge.mm:236] Initializing <RCTCxxBridge: 0x7ff8cbf38af0> (parent: <RCTBridge: 0x600000f79960>, executor: (null))
2020-05-18 15:44:57:381 Newspack[12261:14611562] 🛑 ERROR: /CoreDataManager.swift handleSaveError(_:in:): Unresolved Core Data save error: Error Domain=NSCocoaErrorDomain Code=134030 "An error occurred while saving." UserInfo={NSAffectedObjectsErrorKey=(
    "<StagedEdits: 0x60000070eb70> (entity: StagedEdits; id: 0x60000248e420 <x-coredata:///StagedEdits/tBE1B54B5-AA0C-463C-A0B5-B113E15C274049>; data: {\n    content = nil;\n    excerpt = nil;\n    lastModified = \"1982-02-12 16:00:00 +0000\";\n    postItem = nil;\n    title = nil;\n})"
)}
2020-05-18 15:44:57:382 Newspack[12261:14611562] 🛑 ERROR: /CoreDataManager.swift handleSaveError(_:in:): Generating exception with reason:
Unclassified save error - something we depend on returned an error on null.null
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Notes:**
1. WordPressUI update is needed by WordPressKit. 
2. WordPressKit lives in Authenticator and the main project, so both need updated at the same time.
3. Authenticator, AlamofireImage, and the main project rely on Alamofire, so all three needed updated at the same time.

mindgraffiti committed:
Update the WordPressUI pod to 1.6.0, as it is a prerequisite for Auth

Authenticator 1.16.0-beta requires WordPressUI 1.6.0 as its minimum dependency. Has to do with Swift 5 support.

mindgraffiti committed:
Update interdependent pods

WordPressKit to 4.8.0. Alamofire to 4.8.0. WordPressAuthenticator to 1.16.0-beta.5

mindgraffiti committed:
Update AlamofireImage to 3.5.2, which is compatible with Alamofire 4.8.0

mindgraffiti committed:
Gridicons - remove deprecated method warning

mindgraffiti committed:
Remove warning from Folders nav bar - add button was plain style

It's now bordered style.

aerych reviewed (approved):
Tested the branch. Looking good! :shipit: 
Thanks for wrangling this @mindgraffiti !

Also thanks for spotting the crash. I'm guessing its a regression stemming from when a bunch of obsolete entities and relations were cleaned up. I'll get it sorted!

mindgraffiti commented:
Thanks @aerych!

mindgraffiti committed:
Make Authenticator version conform to version standards

Cocoapods still adopted the correct version, but the version tag should conform to the rest
